sys/net/network_layer/ng_sixlowpan/frag/rbuf.c

```diff
@@ -117,7 +105,7 @@ void rbuf_add(ng_netif_hdr_t *netif_hdr, ng_sixlowpan_frag_t *frag,
     }
 
     while (ptr != NULL) {
-        if (_rbuf_int_in(ptr, offset, offset + dg_frag_size - 1)) {
+        if (_rbuf_int_in(ptr, offset, offset + frag_size - 1)) {
             DEBUG("6lo rfrag: overlapping or same intervals, discarding datagram\n");
             ng_pktbuf_release(entry->pkt);
             _rbuf_rem(entry);
```

Comment:
> What if this pkt has several users?

1. This is not of relevance for this module, since the other user then should have its own reference.
2. Normally this module should be the packet's only user, since it creates it and in this case disregards it because of an error. Only if the packet was reassembled successfully it will be handed over to another thread.